src/api/entities/Account.ts

```diff
@@ -324,20 +324,37 @@ export class Account extends Entity<UniqueIdentifiers, string> {
     const {
       context: {
         polymeshApi: {
-          query: { identity },
+          query: { identity, multiSig },
         },
       },
       context,
       address,
     } = this;
 
-    const identityId = await identity.keyToIdentityIds(stringToAccountId(address, context));
+    const optKeyRecord = await identity.keyRecords(stringToAccountId(address, context));
 
-    if (identityId.isEmpty) {
+    if (optKeyRecord.isEmpty) {
       return null;
     }
 
-    const did = identityIdToString(identityId);
+    const keyRecord = optKeyRecord.unwrap();
+
+    let did: string;
+    if (keyRecord.isPrimaryKey) {
+      did = identityIdToString(keyRecord.asPrimaryKey);
+    } else if (keyRecord.isSecondaryKey) {
+      did = identityIdToString(keyRecord.asSecondaryKey[0]);
+    } else if (keyRecord.isMultiSigSignerKey) {
+      const multiSigAddress = keyRecord.asMultiSigSignerKey;
+      const rawMultiSigDid = await multiSig.multiSigToIdentity(multiSigAddress);
+      did = identityIdToString(rawMultiSigDid);
+    } else {
+      throw new PolymeshError({
```

Comment:
What I really want is exhaustive enum matching.

It might be possible with `never`, but I think this is simpler. Seems like Sonar cloud doesn't like it though, so maybe the multiSig case could be assumed in the `else`.